### PR TITLE
ui: Display contact information in jobs detail page

### PR DESCRIPTION
**Pull request by monaawi**
* Added contact information
* Changed the layout of the jobs detail page
* INSPIR-2339

### CODE REVIEW

File: ui/src/jobs/components/RanksList.jsx [hunk 1]
```diff
@@ -3,13 +3,20 @@ import PropTypes from 'prop-types';
 import { List } from 'immutable';
 
 import InlineList from '../../common/components/InlineList';
+import { RANK_VALUE_TO_DISPLAY } from '../../common/constants';
+// import { rankOptions } from '../../constants';
```
Comment (harunurhan) [inline ui/src/jobs/components/RanksList.jsx]: `leftover`

File: ui/src/jobs/components/ReferenceLetters.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Map } from 'immutable';
+import InlineUL from '../../common/components/InlineList/InlineUL';
+import URLList from '../../common/components/URLList';
+import EmailList from './EmailList';
+
+class ReferenceLetters extends Component {
```
Comment (harunurhan) [inline ui/src/jobs/components/ReferenceLetters.jsx]: `ReferenceLetterContacts`

File: ui/src/jobs/components/ContactList.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+class ContactList extends Component {
+  static renderContact(contact) {
+    const email = contact.get('email');
+    const name = contact.get('name');
+    const renderParanthesis = name && email;
+    return (
+      <>
+        {name && (
+          <>
```
Comment (harunurhan) [inline ui/src/jobs/components/ContactList.jsx]: reduntant

File: ui/src/jobs/components/ReferenceLetters.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  render() {
+    const { referenceLetters } = this.props;
+    const urls = referenceLetters.get('urls', null);
+    const emails = referenceLetters.get('emails', null);
```
Comment (harunurhan) [inline ui/src/jobs/components/ReferenceLetters.jsx]: `const shouldRender = urls || emails` 

that way you get rid of warnings that may appear when you remove default `null` for `get`.

File: ui/src/jobs/components/ContactList.jsx (hunk 3 above)
Comment (monaawi) [reply ui/src/jobs/components/ContactList.jsx]: For this, I couldn't add the space without adding the fragment, I guess I'm doing it wrong

File: ui/src/jobs/components/ContactList.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import InlineList from '../../common/components/InlineList';
+
+class ContactList extends Component {
+  static renderContact(contact) {
+    const email = contact.get('email');
+    const name = contact.get('name');
+    const renderParanthesis = name && email;
+    return (
+      <>
+        {name && <span>{`${contact.get('name')}`}</span>}
```
Comment (drjova) [inline ui/src/jobs/components/ContactList.jsx]:
```suggestion
        {name && <span>{name}</span>}
```

File: ui/src/jobs/components/ContactList.jsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        {renderParanthesis && ' ('}
+        {email && (
+          <ExternalLink href={`mailto:${contact.get('email')}`}>
+            {contact.get('email')}
```
Comment (drjova) [inline ui/src/jobs/components/ContactList.jsx]:
```suggestion
            {email}
```

File: ui/src/jobs/components/__tests__/MoreInfo.test.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        value: 'url1',
+      },
+      {
+        value: 'url2',
+      },
+    ]);
+    const wrapper = shallow(<MoreInfo urls={urls} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+});
```
Comment (drjova) [inline ui/src/jobs/components/__tests__/MoreInfo.test.jsx]: what if 

```js
<MoreInfo />
```

File: ui/src/jobs/components/DeadlineDate.jsx [hunk 8]
```diff
@@ -1,12 +1,19 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import moment from 'moment';
 import IconText from '../../common/components/IconText';
 
 class DeadlineDate extends Component {
   render() {
     const { deadlineDate } = this.props;
+    const formattedDeadlineDate = moment(deadlineDate).format('MMM D, YYYY');
```
Comment (drjova) [inline ui/src/jobs/components/DeadlineDate.jsx]: I'd move this to a ``static``

File: ui/src/jobs/components/__tests__/MoreInfo.test.jsx (hunk 7 above)
Comment (drjova) [reply ui/src/jobs/components/__tests__/MoreInfo.test.jsx]: https://github.com/inspirehep/inspirehep/pull/212/files#diff-545b14684e0ea3d99a7df1f6b0ecd2d2R24

File: ui/src/jobs/components/__tests__/__snapshots__/ContactList.test.jsx.snap [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+exports[`ContactList renders with contacts with both email and name 1`] = `
+<div
+  className="__InlineList__"
+>
+  <ul
+    className="separate-items-with-comma"
+  >
+    <li
+      key="Map { \\"email\\": \\"johndoe@yahoo.com\\", \\"name\\": \\"John\\" }"
```
Comment (drjova) [inline ui/src/jobs/components/__tests__/__snapshots__/ContactList.test.jsx.snap]: I'd use ``email`` as a key

File: ui/src/jobs/components/__tests__/MoreInfo.test.jsx (hunk 7 above)
Comment (monaawi) [reply ui/src/jobs/components/__tests__/MoreInfo.test.jsx]: As in add a test with <MoreInfo/> ? The snapshot would be empty for this, does it make sense?

File: ui/src/jobs/components/DeadlineDate.jsx (hunk 8 above)
Comment (monaawi) [reply ui/src/jobs/components/DeadlineDate.jsx]: what would be the advantage of this?

File: ui/src/jobs/components/__tests__/MoreInfo.test.jsx (hunk 7 above)
Comment (drjova) [reply ui/src/jobs/components/__tests__/MoreInfo.test.jsx]: if urls is empty, doesn't the https://github.com/inspirehep/inspirehep/pull/212/files#diff-545b14684e0ea3d99a7df1f6b0ecd2d2R21 complain?

File: ui/src/jobs/components/__tests__/__snapshots__/ContactList.test.jsx.snap (hunk 9 above)
Comment (monaawi) [reply ui/src/jobs/components/__tests__/__snapshots__/ContactList.test.jsx.snap]: actually, email or name because both are optional and it will render with one or the other

File: ui/src/jobs/components/DeadlineDate.jsx (hunk 8 above)
Comment (drjova) [reply ui/src/jobs/components/DeadlineDate.jsx]: just to be consistent, we usually use ``static`` for calculated params

Comment (monaawi) [reply ui/src/jobs/components/DeadlineDate.jsx]: I wouldn't make it static in this case as that won't add any value and it is also quite small, just a date format

Comment (drjova) [reply ui/src/jobs/components/DeadlineDate.jsx]: It's just a suggestion, you did it already for small things like ``renderRank`` https://github.com/inspirehep/inspirehep/pull/212/files#diff-f1c078865507d0e537db58f068149762R9 I don't see why it's different

Comment (monaawi) [reply ui/src/jobs/components/DeadlineDate.jsx]: It's different because renderRank needs to be passed to the InLineList
